hack/core/repositories/version_repository.py

```
@@ -0,0 +1,21 @@
+        self.yaml: YAML = get_yaml_instance()
+
+    def find_all(self) -> list[Version]:
+        if not os.path.isfile(self.file_path):
+            return []
+        with open(self.file_path, "r") as f:
+            data = self.yaml.load(f)
+            try:
+                parsed = VersionsFile(**data)
+                return parsed.versions
+            except Exception as e:
+                raise ValueError(f"Invalid versions.yaml structure: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)